src/main/java/com/strandls/esmodule/services/impl/ElasticSearchServiceImpl.java

```diff
@@ -530,7 +530,19 @@ public AggregationResponse aggregation(String index, String type, MapSearchQuery
 		BoolQueryBuilder masterBoolQuery = getBoolQueryBuilder(searchQuery);
 
 		applyMapBounds(searchParams, masterBoolQuery, geoAggregationField);
-		AggregationBuilder aggregation = AggregationBuilders.terms(filter).field(filter).size(1000);
+
+		AggregationBuilder aggregation;
+
+		if (filter.equals("max_voted_reco.scientific_name.keyword")) {
```

Comment:
use max_voted_reco... as final constant instead of a String

Reply:
set MVR_SCIENTIFIC_NAME equal to this string value in the Constants file and used here.